app/src/main/java/org/fossasia/openevent/general/event/TicketService.kt

```diff
@@ -1,10 +1,29 @@
 package org.fossasia.openevent.general.event
 
-import io.reactivex.Flowable
+import io.reactivex.Single
 
-class TicketService(private val ticketApi: TicketApi) {
+class TicketService(private val ticketApi: TicketApi, private val ticketsDao: TicketsDao) {
 
-    fun getTickets(id: Long): Flowable<List<Ticket>> {
-        return ticketApi.getTickets(id)
+    fun getTickets(id: Long): Single<List<Ticket>> {
+        val ticketsFLowable = ticketsDao.getAllTickets(id)
+        return ticketsFLowable.flatMap{
+            if (it.isNotEmpty())
+                ticketsFLowable
+            else
+                ticketApi.getTickets(id)
+                        .map {
+                            setEventId(it, id)
+                            ticketsDao.insertTickets(it)
+                        }
+                        .flatMap {
+                            ticketsFLowable
+                        }
+        }
+    }
+
+    fun setEventId(tickets: List<Ticket>, id: Long){
+        for(ticket: Ticket in tickets){
```

Comment:
Here I am setting the event id, like the id of the event related with the tickets to each ticket